smarts/sstudio/genscenario.py

```diff
@@ -54,95 +149,205 @@ def gen_scenario(
+            if not road_map:
+                logger.warning(
+                    "No reference to a RoadNetwork file was found in {}, or one could not be created. "
+                    "Please make sure the path passed is a valid Scenario with RoadNetwork file required "
+                    "(or a way to create one) for scenario building.".format(
+                        scenario_dir
+                    )
+                )
+                return
+
+            map_dir = os.path.join(build_dir, "map")
+            os.makedirs(map_dir, exist_ok=True)
+            road_map.to_glb(os.path.join(map_dir, "map.glb"))
+            _update_artifacts(db_conn, artifact_paths, map_hash)
+
+    # Traffic
+    artifact_paths = build_graph["traffic"]
+    obj_hash = pickle_hash(scenario.traffic, True)
+    if _needs_build(
+        db_conn, scenario.traffic, artifact_paths, obj_hash, map_needs_rebuild
+    ):
+        with timeit("traffic", logger.info):
+            for name, traffic in scenario.traffic.items():
+                _gen_traffic(
+                    scenario=scenario_dir,
+                    traffic=traffic,
+                    name=name,
+                    seed=seed,
+                    overwrite=True,
+                    map_spec=map_spec,
+                )
+            _update_artifacts(db_conn, artifact_paths, obj_hash)
+
+    # Ego missions
+    artifact_paths = build_graph["ego_missions"]
+    obj_hash = pickle_hash(scenario.ego_missions, True)
+    if _needs_build(
+        db_conn, scenario.ego_missions, artifact_paths, obj_hash, map_needs_rebuild
+    ):
+        with timeit("ego_missions", logger.info):
+            missions = []
+            for mission in scenario.ego_missions:
+                if isinstance(mission, types.GroupedLapMission):
+                    _gen_group_laps(
+                        scenario=output_dir,
+                        begin=mission.route.begin,
+                        end=mission.route.end,
+                        grid_offset=mission.offset,
+                        used_lanes=mission.lanes,
+                        vehicle_count=mission.actor_count,
+                        num_laps=mission.num_laps,
+                        seed=seed,
+                        overwrite=True,
+                        map_spec=map_spec,
+                    )
+                else:
+                    missions.append(mission)
+
+            if missions:
+                _gen_agent_missions(
                     scenario=output_dir,
-                    begin=mission.route.begin,
-                    end=mission.route.end,
-                    grid_offset=mission.offset,
-                    used_lanes=mission.lanes,
-                    vehicle_count=mission.actor_count,
-                    num_laps=mission.num_laps,
+                    missions=missions,
                     seed=seed,
-                    overwrite=overwrite,
+                    overwrite=True,
```

Comment:
Does the `overwrite` parameter make sense anymore for the `_gen` methods? It might just be reasonable to assume `True` always and remove the parameter.

Reply:
Yeah I agree, I'll make this change.